examples/scripts/TreeCompareAuto.C

```diff
@@ -0,0 +1,309 @@
+ *              This software is distributed under the terms of the             *
+ *              GNU Lesser General Public Licence (LGPL) version 3,             *
+ *                  copied verbatim in the file "LICENSE"                       *
+ ********************************************************************************/
+
+// The macro compares all the leaves of two TTree objects.
+// The first and strongest comparison is that the entries in the two leaves
+// to compare  are idential.
+// To check this one loops over the entries in the trees, calculate the
+// difference between the two values and fills the result in a histogram.
+// If all entries are absolutely identical the result in the histogram is a
+// delta function at 0.
+// If the first check for identical entries fails it is checked if there are
+// only some changes in the order of the entries. In this case the two
+// produced histgrams (one for each tree) are identical.
+// The last check which again is only done if both previous test fails uses a
+// Kolmogorov test to check if the produced hostograms are comparable on a
+// statistical base.
+// If at least for one leaf all theree comparisons fail the complete test
+// fails.
+
+#include <RtypesCore.h>
+#include <TBranch.h>
+#include <TFile.h>
+#include <TH1.h>
+#include <TMath.h>
+#include <TObjArray.h>
+#include <TString.h>
+#include <TTree.h>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <utility>
+#include <vector>
+
+std::vector<std::pair<TString, TString>> GetLeafNames(TTree*);
+
+Bool_t CheckEntriesIdentical(TH1* compHist, std::stringstream& outstream);
+Bool_t CheckHistogramIdentical(TH1* origHist, TH1* newHist, std::stringstream& outstream);
+Bool_t CheckHistogramKolmogorov(TH1* origHist, TH1* newHist, std::stringstream& outstream);
+
+int TreeCompareAuto(TString fileName1 = "", TString fileName2 = "")
+{
+    if (fileName1.IsNull() || fileName2.IsNull()) {
+        cout << "Filenames are not defined" << endl;
+        exit(42);
+    }
+
+    // Get the output tree from the original file
+    TFile* file1 = TFile::Open(fileName1, "READ");
+    if (nullptr == file1)
+        return 42;
+    TTree* tree1 = (TTree*)file1->Get("cbmsim");
+
+    // Get the output tree from the file whcih should be compared
+    TFile* file2 = TFile::Open(fileName2, "READ");
+    if (nullptr == file2)
+        return 42;
+    TTree* tree2 = (TTree*)file2->Get("cbmsim");
+
+    // Add the output tree from the file to compare as friend to the tree
+    // of the original file. This allows to access a data member of the
+    // original file by e.g. StsHit.fX and the data element of the second tree
+    // by tree2.StsHit.fX
+    tree1->AddFriend(tree2, "tree2");
+
+    // Define pairs of data members to compare. One from each tree.
+    // This allows to compare them also if names or the structure of
+    // the classses change.
+    std::vector<std::pair<TString, TString>> leaves = GetLeafNames(tree1);
```

Comment:
```suggestion
    auto const leafNames = GetLeafNames(tree1);

    // Define pairs of data members to compare. One from each tree.
    // This allows to compare them also if names or the structure of
    // the classes change.    
    std::vector<std::pair<TString, TString>> leaves;
    for (auto const leafName : leafNames) {
        TString nameTree2 = "tree2." + leafName;
        leaves.emplace_back(make_pair(leafName, nameTree2));
    }
```

* See [my comment down below](https://github.com/FairRootGroup/FairRoot/pull/1419/files#r1247871827)

Reply:
Fixed. The second name can even be generated on the fly.